src/factories/flowRunArtifact.ts

```diff
@@ -31,10 +44,18 @@ export async function flowRunArtifactFactory(artifact: Artifact) {
       return
     }
 
+    let selectedOffset = 0
```

Comment:
Again, simply tracking whether the selected border is visible or not for the artifact, then adjusting it's placement for consistency.

Reply:
It seems like the artifactCluster and flowRunArtifact share a lot of the same logic. I haven't dug into it but seems like they could follow a pattern similar to nodes where there's a single node factory that can be a flow run or a task run. But they share functionality.